alexferl/flask-mysqldb
# fix: gracefully handle MySQLdb.OperationalError in teardown

### Description
This pull request improves the `teardown` method in `flask_mysqldb` by adding error handling for `MySQLdb.OperationalError`. Previously, when `ctx.mysql_db.close()` failed with an error such as `(2006, '')`, it could disrupt the teardown process without providing helpful feedback.

### Changes Made
- Added a `try-except` block to the `teardown` method to catch `MySQLdb.OperationalError`.
- Logged the error using `current_app.logger.error` for better visibility and debugging.

### Why This Change?
The error `(2006, '')` occurs when the MySQL server has gone away, leading to unhandled exceptions during request teardown. This fix ensures that such errors are handled gracefully, without interrupting the application flow, while still providing useful logs for developers.

### How to Test
1. Trigger a scenario where `ctx.mysql_db.close()` raises `MySQLdb.OperationalError` (e.g., simulate a "MySQL server has gone away" error).
2. Verify that the application does not crash and that the error is logged properly in the application's logs.

### Notes
Let me know if any further modifications or tests are required!
